src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -891,21 +892,49 @@ else if (v instanceof ListValue)
             throw new InternalExpressionException("'effect' needs either no arguments (clear) or effect name, duration, and optional amplifier, show particles and show icon");
         });
 
-        // gamemode
-        // spectate
-        // "fire"
-        // "extinguish"
-        // "silent"
-        // "gravity"
-        // "invulnerable"
-        // "dimension"
-        // "item"
-        // "count",
-        // "age",
-        // "effect_"name
-        // "hold"
-        // "hold_offhand"
-        // "jump"
+        put("gamemode", (e,v)->{
+            if(!(e instanceof ServerPlayerEntity)){
+                throw new InternalExpressionException("Cannot change the gamemode of non-player entities");
+            }
+            
+            switch(v.getString().toLowerCase()){
+                case "survival":((ServerPlayerEntity) e).setGameMode(GameMode.SURVIVAL);return;
+                case "creative":((ServerPlayerEntity) e).setGameMode(GameMode.CREATIVE);return;
+                case "adventure":((ServerPlayerEntity) e).setGameMode(GameMode.ADVENTURE);return;
+                case "spectator":((ServerPlayerEntity) e).setGameMode(GameMode.SPECTATOR);return;
+            }
+
+            switch((int)NumericValue.asNumber(v).getLong()){
+                case 0:((ServerPlayerEntity) e).setGameMode(GameMode.SURVIVAL);return;
+                case 1:((ServerPlayerEntity) e).setGameMode(GameMode.CREATIVE);return;
+                case 2:((ServerPlayerEntity) e).setGameMode(GameMode.ADVENTURE);return;
+                case 3:((ServerPlayerEntity) e).setGameMode(GameMode.SPECTATOR);return;
+            }
+            throw new InternalExpressionException(NumericValue.asNumber(v).getLong()+" is an invalid gamemode id");
+        });
+
+        put("jumping",(e,v)->{
```

Comment:
I would still add jump(), just get access to that method through LivingEntityInterface

Reply:
as put("jump", ....)

Reply:
what is LivingEntityInterface?

Reply:
I did it, but I had to do it by basically copy pasting code from the jump() mthod in LivingEntity.java, then copying abit of code form other places which was also protected and private, and finally got something that may or may not work

Reply:
at least it works...